Sources/PackageGraph/ModulesGraph.swift

```diff
@@ -207,7 +207,8 @@ public struct ModulesGraph {
                         switch dependency {
                         case .target(let targetDependency, _):
                             allTargets.insert(targetDependency)
-                            modulesToPackages[targetDependency.id] = package
+                            modulesToPackages[targetDependency.id] =
+                                identitiesToPackages[targetDependency.packageIdentity]
```

Comment:
If we change `ResolvedModule` and `ResolvedProduct` to reference `ResolvedPackage` instead of `PackageIdentifier` that should make `modulesToPackages` and `productsToPackages` obsolete...